src/GafferSceneModule/AttributesBinding.cpp

```diff
@@ -53,10 +54,84 @@
 #include "GafferScene/StandardAttributes.h"
 
 #include "GafferBindings/DependencyNodeBinding.h"
+#include "GafferBindings/Serialisation.h"
+
+#include <cassert>
 
 using namespace boost::python;
 using namespace GafferScene;
 
+namespace AttributeQueryBinding
+{
+
+bool isSetup( AttributeQuery const& query )
+{
+	IECorePython::ScopedGILRelease gilRelease;
+	return query.isSetup();
+}
+
+bool canSetup( AttributeQuery const& query, Gaffer::Plug const& plug )
+{
+	IECorePython::ScopedGILRelease gilRelease;
+	return query.canSetup( & plug );
+}
+
+void setup( AttributeQuery& query, Gaffer::Plug const& plug )
+{
+	IECorePython::ScopedGILRelease gilRelease;
+	if( !( query.canSetup( & plug ) ) )
+	{
+		throw IECore::Exception( "AttributeQuery cannot be setup from specified plug" );
+	}
```

Comment:
I did it this way to avoid a call to canSetup() from setup() in the C++ code. In the python binding wrapper function for setup() there is a call to canSetup() before the call to setup()